### DISCUSSION TITLE
feature(sheet): options to have "fit" mode with a max-height

### DISCUSSION THREAD

**Discussion by idrakimuhamad** [tamagui/tamagui#1687] [General]
The `fit` mode was great, I was doing my own layout calculation before to feed `Sheet` with the proper height to it so I can get that behaviour, so it's awesome to have it built in.

However, I'd also think that it would be fitting to have this option to work in tandem with a percentage or height, which would be the biggest point instead of the `fit` to be the biggest snap points.

One example is that for a list within a sheet. For a short list, I'd like to open up to the content size. And when the list is bigger than certain breakpoint, I'd like the sheet to max to that size, instead of overflowing bigger than the screen size.

Right now I have done some workaround -- I'm using a `mixed` mode, and put the snap points into a state. I've also put a layout event on my element just inside the `Sheet.Frame` and calculate its layout and if its more than the percentage that I'd like, I'd set the snap points to percentage eg 80%, else I'd set it to `fit`. This seems to work decent, but I'm not sure if its performant, and I do sometimes get some error thrown by Animated eg `Excessive number of pending callbacks` which doesn't happen before adding the layout event.

Would be great if there's a prop for this perhaps somewhere on the layout event within the Sheet, that could do such thing.

**Comment by nderscore** [comment]
Right now there's a hard coded maximum for `fit` mode that caps out at `100%` of the screen height (or parent container height in non-modal mode)

Until there's a way to configure/control that behavior, if you want a simpler workaround you can set a maximum height on the `Sheet.Frame` in pixels (percentage heights don't work because of the additional wrapper elements) like this:
```tsx
const { height } = useWindowDimensions();

<Sheet snapPointsMode="fit">
  <Sheet.Frame maxHeight={height * .8}> // constrain height to maximum of 80% screen height
    {/* ... */}
  </Sheet.Frame>
</Sheet>

**Comment by natew** [comment]
@nderscore did your latest change address this or no? Converting to a discussion for now.

**Reply by nderscore** [reply]
Not really. I just happened to notice a bug while I was playing around with mixed mode. Still thinking about the best way to support this use case along with safe areas.
